topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -316,107 +354,70 @@ Dynamic destinations allow you to write custom python code to dispatch jobs base
+on settings that have worked well in the usegalaxy.* federation. The rule file can simply be imported directly, with local overrides applied on top.
 
-> <hands-on-title>Writing a DTD</hands-on-title>
->
-> 1. Dynamic tool destinations are configured via a YAML file. As before, we'll use a fake example but this is extremely useful in real-life scenarios. Create the file `templates/galaxy/config/tool_destinations.yml` with the following contents:
->
->    {% raw %}
->    ```diff
->    --- /dev/null
->    +++ b/templates/galaxy/config/tool_destinations.yml
->    @@ -0,0 +1,11 @@
->    +---
->    +tools:
->    +  testing:
->    +    rules:
->    +      - rule_type: file_size
->    +        lower_bound: 16
->    +        upper_bound: Infinity
->    +        destination: slurm-2c
->    +    default_destination: slurm
->    +default_destination: slurm
->    +verbose: True
->    {% endraw %}
->    ```
->    {: data-commit="Add tool destinations conf"}
->
->    The rule says:
->    - If the tool has ID `testing`:
->      - If the input dataset is >=16 bytes, run on the destination `slurm-2c`
->      - If the input dataset is <16 bytes, run on the destination `slurm`
->    - Else, run on the destination `slurm`
+> <hands-on-title>Adding the TPV shared database to job_conf</hands-on-title>
 >
-> 2. We also need to inform Galaxy of the path to the file we've just created, which is done using the `tool_destinations_config_file` in `galaxy_config` > `galaxy`. Additionally we need to add a `galaxy_config_templates` entry to ensure it is deployed.
+> 1. Edit `group_vars/galaxyservers.yml` and add the location of the TPV shared rule file to the `tpv_dispatcher` destination.
 >
 >    {% raw %}
 >    ```diff
 >    --- a/group_vars/galaxyservers.yml
 >    +++ b/group_vars/galaxyservers.yml
->    @@ -113,6 +113,8 @@ galaxy_config:
->         # Data Library Directories
->         library_import_dir: /libraries/admin
->         user_library_import_dir: /libraries/user
->    +    # Tool Destination Configuration
->    +    tool_destinations_config_file: "{{ galaxy_config_dir }}/tool_destinations.yml"
->       gravity:
->         process_manager: systemd
->         galaxy_root: "{{ galaxy_root }}/server"
->    @@ -141,6 +143,8 @@ galaxy_config_templates:
->         dest: "{{ galaxy_config.galaxy.containers_resolvers_config_file }}"
->       - src: templates/galaxy/config/dependency_resolvers_conf.xml
->         dest: "{{ galaxy_config.galaxy.dependency_resolvers_config_file }}"
->    +  - src: templates/galaxy/config/tool_destinations.yml
->    +    dest: "{{ galaxy_config.galaxy.tool_destinations_config_file }}"
->     
->     galaxy_local_tools:
->     - testing.xml
+>    @@ -40,6 +40,7 @@ galaxy_config:
+>                 function: map_tool_to_destination
+>                 rules_module: tpv.rules
+>                 tpv_config_files:
+>    +              - https://raw.githubusercontent.com/galaxyproject/tpv-shared-database/main/tools.yml
```

Comment:
how often does this get pulled? is it cached?

Reply:
Pulled once on each Galaxy startup.